keyboards/keychron/q11/q11.c

```diff
@@ -0,0 +1,124 @@
+/* Copyright 2023 @ Keychron (https://www.keychron.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "quantum.h"
+
+// Mask out handedness diode to prevent it
+// from keeping the keyboard awake
+// - just mirroring `KC_NO` in the `LAYOUT`
+//   macro to keep it simple
+const matrix_row_t matrix_mask[] = {
+    0b011111111,
+    0b011111111,
+    0b011111111,
+    0b001111111,
+    0b011111101,
+    0b001011111,
+    0b111111111,
+    0b101111111,
+    0b111111111,
+    0b110111111,
+    0b010111111,
+    0b111011110,
+};
+
+#ifdef DIP_SWITCH_ENABLE
+bool dip_switch_update_kb(uint8_t index, bool active) {
+    if (!dip_switch_update_user(index, active)) {
+        return false;
+    }
+    if (index == 0) {
+        default_layer_set(1UL << (active ? 0 : 2));
+    }
+    return true;
+}
+#endif
+
+#ifdef ENCODER_ENABLE
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+    if (!encoder_update_user(index, clockwise)) {
+      return false; /* Don't process further events if user function exists and returns false */
+    }
+    if (index == 0) { /* First encoder */
+        if (clockwise) {
+            tap_code_delay(KC_VOLU, 10);
+        } else {
+            tap_code_delay(KC_VOLD, 10);
+        }
+    } else if (index == 1) { /* Second encoder */
+        if (clockwise) {
+            tap_code_delay(KC_VOLU, 10);
+        } else {
+            tap_code_delay(KC_VOLD, 10);
+        }
+    }
+    return true;
+}
+#endif
```

Comment:
Since https://github.com/qmk/qmk_firmware/pull/20320 is merged into `develop` now, the use of this code won't be needed for when `develop` gets merged into `master` since in [`quantum/encoders.c`](https://github.com/waffle87/qmk_firmware/blob/654235f949a49d2e26e2ce8a581784b3cc7a7e34/quantum/encoder.c#L79-L103), the fallback is there, so a PR to remove this code would suffice when the time comes.

Reply:
Thanks, but we have decided to delete it now to avoid submitting another PR.ha ha ha....

Reply:
Probably for the best anyways, just to save the headache of another PR😄